### DISCUSSION TITLE
Stage parameters not being resolved

### DISCUSSION THREAD

**Discussion by markyarnold** [serverless/serverless#11464] [Q&A]
Hi all;  Hope you're all well.

I have a simple scaffold that I'm creating.  But, I cant get the params to resolve.  They are defined for the dev stage - but on deployment they are not resolving:

- Cannot resolve variable at "provider.environment.SG_ID": The param "sgId" cannot be resolved......... for example

![image](https://user-images.githubusercontent.com/56762387/197338568-1594b41c-1451-473e-afbc-cd80af08602e.png)

Any advice very gratefully received.

**Comment by ShlomoVinny** [comment]
If what youre trying to do is pass a SubnetId and SecurityGroupId from your VPC to your lambdas, there are better ways to achieve that.

Using the [serverless-vpc-plugin](https://www.npmjs.com/package/serverless-vpc-plugin) to create the said VPC with serverless in the first place, which will automagically connect your serverless lambdas to said VPC.
Or [vpc-discovery-plugin](https://www.npmjs.com/package/serverless-vpc-discovery) to connect an existing VPC to your serverless lambdas. Or vise versa.
